### ISSUE TITLE
Artemis should do all the work between grid detection and grids

### ISSUE THREAD

**Issue by DominicOram** [DiamondLightSource/python-artemis#567]
Currently we have some prototype grid detection code but before we can use it there are some bits that GDA is doing between grid detection and performing the grid scan, namely, at least:
* Move the backlight out
* Move to data collection on the scatterguard/aperture

There are also some bits that GDA is doing here that we could just move earlier in GDA for now:
* Topup gate
* Opening detector shutter
* Preparing sample id for sending to ispyb

## Acceptance Criteria
* There is a new plan/endpoint on Artemis that does the required work between grid detection and grid scan
* GDA is refactored so that it does grid detection and immediately launches this plan

**Comment by DominicOram** [comment]
Closing in favour of https://github.com/DiamondLightSource/python-artemis/issues/652
